### ISSUE TITLE
EXC_BREAKPOINT void JSC::JSLock::unlock(intptr_t)

### ISSUE THREAD

**Issue by matinzd** [software-mansion/react-native-screens#1309]
## Description
Found it in our crash logs. 

### Screenshots
<img width="592" alt="Screenshot 2022-02-09 at 15 36 31" src="https://user-images.githubusercontent.com/24797481/153223064-b1c15eea-5aa6-423d-ac15-bd723e641ba7.png">

## Steps To Reproduce
It's happening on our release build on iOS. We don't have any idea how it's happening.

## Reproduction
It's happening on our release build on iOS. We don't have any idea how it's happening.

## Platform
- [x] iOS
- [ ] Android
- [ ] Web
- [ ] Windows
- [ ] tvOS

## Workflow

- [ ] Managed workflow
- [ ] Bare workflow

## Package versions

| package                                | version |
| -------------------------------------- | ------- |
| react-native                           |     0.67.2    |
| @react-navigation/native               |    6.0.6     |
| @react-navigation/native-stack         |    6.2.5     |
| react-native-screens                   |   3.10.1      |
| react-native-safe-area-context         |    3.2.2     |
| react-native-gesture-handler           |     2.1.0    |
| react-native-reanimated                |    2.3.1     |
| expo                                   |    N/A     |

Crash report log:
[e1899330444b4951adebe70d3253925d-symbolicated.log](https://github.com/software-mansion/react-native-screens/files/8033519/e1899330444b4951adebe70d3253925d-symbolicated.log)

**Comment by kacperkapusciak** [comment]
Hi @matinzd,

From the crash, I can't deduce anything directly connected to `screens` library. Unfortunately, it's a very generic crash coming from iOS's JavaScript virtual machine.

Not sure how Sentry labels the crashes and why it has chosen `RNSScreen` as a source of the crash. Maybe it just so happened that `RNSScreen` was "the closes thing" in code to the crash but it could've occurred a few layers of abstraction deeper in the VM.

TL;DR without more information this can be anything from nothing to everything 😕

**Comment by matinzd** [comment]
@kacperkapusciak I am also not sure what is happening there and I couldn't reproduce it in development. Maybe Sentry is just labeling it by mistake. It's really a deep OS-level crash and I understand. Will let you know if I could reproduce it.

**Comment by pistonsky** [comment]
The RNSScreen label has nothing to do with it. I have the same crash labelled UISystemKeyboardDockController.

I was able to catch it while attached to Xcode, and it lead to some breakpoint binary instruction. The last code was in `JSObjectCopyPropertyNames` which belongs to react-native's `ReactCommon/jsi/JSCRuntime.cpp`:
```
// Very expensive
jsi::Array JSCRuntime::getPropertyNames(const jsi::Object &obj) {
  JSPropertyNameArrayRef names =
      JSObjectCopyPropertyNames(ctx_, objectRef(obj)); // <-- happens on this line
  size_t len = JSPropertyNameArrayGetCount(names);
  // Would be better if we could create an array with explicit elements
  auto result = createArray(len);
  for (size_t i = 0; i < len; i++) {
    JSStringRef str = JSPropertyNameArrayGetNameAtIndex(names, i);
    result.setValueAtIndex(*this, i, createString(str));
  }
  JSPropertyNameArrayRelease(names);
  return result;
}
```
If I hit "continue execution" it will stop on the same line again and again. So it's an infinite loop there.

I went up in the stack of that thread and discovered that it happens when some shared value gets initialised with some complex object. That means the solution would be to try to simplify values that you put into a shared value.

**Comment by kacperkapusciak** [comment]
Wow, thanks @pistonsky for the detailed description :tada:

So, as we've been guessing Sentry is referencing the closest thing in the code to the crash - here - the closest controller.

As the issue has nothing to do with `react-native-screens` I'll close the issue. I encourage you to report the issue directly in the [react-native](https://github.com/facebook/react-native) repo and reference this issue as we can't really help with this.

Cheers
